lib/chef/provider/windows_task.rb

```diff
@@ -0,0 +1,411 @@
+        }
+
+        Chef::Log.debug "looking for existing tasks"
+
+        # we use shell_out here instead of shell_out! because a failure implies that the task does not exist
+        xml_cmd = shell_out("schtasks /Query /TN \"#{@new_resource.task_name}\" /XML")
+
+        return if xml_cmd.exitstatus != 0
+
+        doc = REXML::Document.new(xml_cmd.stdout)
+
+        options.each do |option|
+          Chef::Log.debug 'Removing former #{option} if any'
+          doc.root.elements.delete(xml_element_mapping[option])
+          option_value = @new_resource.send("#{option}")
+
+          if option_value
+            Chef::Log.debug "Setting #option as #option_value"
+            split_xml_path = xml_element_mapping[option].split("/") # eg. if xml_element_mapping[option] = "Actions/Exec/WorkingDirectory"
+            element_name = split_xml_path.last # element_name = "WorkingDirectory"
+            cwd_element = REXML::Element.new(element_name)
+            cwd_element.add_text(option_value)
+            element_root = (split_xml_path - [element_name]).join("/") # element_root = 'Actions/Exec'
+            exec_element = doc.root.elements[element_root]
+            exec_element.add_element(cwd_element)
+          end
+        end
+
+        temp_task_file = ::File.join(ENV["TEMP"], "windows_task.xml")
+        begin
+          ::File.open(temp_task_file, "w:UTF-16LE") do |f|
+            doc.write(f)
+          end
+
+          options = {}
+          options["RU"] = @new_resource.user if @new_resource.user
+          options["RP"] = @new_resource.password if @new_resource.password
+          options["IT"] = "" if @new_resource.interactive_enabled
+          options["XML"] = temp_task_file
+
+          run_schtasks("DELETE", "F" => "")
+          run_schtasks("CREATE", options)
+        ensure
+          ::File.delete(temp_task_file)
+        end
+      end
+
+      def load_task_hash(task_name)
+        Chef::Log.debug "Looking for existing tasks"
+
+        task_script = <<-EOH
+          [Console]::OutputEncoding = [Text.UTF8Encoding]::UTF8
+          schtasks /Query /FO LIST /V /TN \"#{task_name}\"
+        EOH
+
+        output = powershell_out("#{task_script}").stdout.force_encoding("UTF-8")
+        if output.empty?
+          task = false
+        else
+          task = {}
+
+          output.split("\n").map! do |line|
+            line.split(": ").map!(&:strip)
+          end.each do |field|
+            if field.is_a?(Array) && field[0].respond_to?(:to_sym)
+              key = (field - [field.last]).join(": ")
+              task[key.gsub(/\s+/, "").to_sym] = field.last
+            end
+          end
+        end
+
+        task_xml = load_task_xml task_name
+        task.merge!(task_xml) if task && task_xml
+
+        task
+      end
+
+      def load_task_xml(task_name)
+        xml_cmd = shell_out("chcp 65001 >nul 2>&1 && schtasks /Query /TN \"#{task_name}\" /XML")
```

Comment:
https://github.com/chef/chef/pull/5886#discussion_r108259818
I also made PR to upstream https://github.com/chef-cookbooks/windows/pull/461